jetbrains-core/src/software/aws/toolkits/jetbrains/core/credentials/RefreshConnectionAction.kt

```diff
@@ -0,0 +1,28 @@
+// Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package software.aws.toolkits.jetbrains.core.credentials
+
+import com.intellij.icons.AllIcons
+import com.intellij.openapi.actionSystem.AnAction
+import com.intellij.openapi.actionSystem.AnActionEvent
+import com.intellij.openapi.project.DumbAware
+import software.aws.toolkits.jetbrains.core.AwsResourceCache
+import software.aws.toolkits.resources.message
+
+class RefreshConnectionAction : AnAction(message("settings.refresh.description"), null, AllIcons.Actions.Refresh), DumbAware {
+
+    override fun update(e: AnActionEvent) {
+        val project = e.project ?: return
+        when (ProjectAccountSettingsManager.getInstance(project).connectionState) {
+            is ConnectionState.InitializingToolkit, is ConnectionState.ValidatingConnection -> e.presentation.isEnabled = false
+            else -> e.presentation.isEnabled = true
+        }
+    }
+
+    override fun actionPerformed(e: AnActionEvent) {
+        val project = e.project ?: return
+        ProjectAccountSettingsManager.getInstance(project).refreshConnectionState()
+        AwsResourceCache.getInstance(project).clear()
```

Comment:
I would clear the cache first since technically this will go and notify the tree to refresh. Ya, likely to never happen but let's make it so it can't

Reply:
yes, you're right.